### PR TITLE
Add additional LayC guidance in docs

### PR DESCRIPTION
**Pull request by John-Clifton-SonarSource**
Adding in additional material and a bit of refactoring for clarity.

## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: docs/description.adoc
Comment (loris-s-sonarsource) [inline docs/description.adoc:1]: Another thing: The PR title does not reflect the intention of the PR

Comment (John-Clifton-SonarSource) [reply docs/description.adoc:1]: Good point. I have made some updates.

File: docs/description.adoc [hunk 1]
```diff
@@ -6,6 +6,24 @@ This document describes how `+rule.adoc+` and its dependencies should be structu
...
+* If you need the information to fix the issue, but only the first time, it probably belongs in ‘Why is this an issue?’
+* If you need the information to fix the issue every time, it probably belongs in ‘How to fix it?’
+* If you don’t need the information to fix the issue, but it will help users grow their knowledge, it probably belongs in ‘Resources’ (this maps to the 'More Info' tab in the products)
+
+By being careful about what goes where, we help to ensure that users get exactly what they need, when they need it.
+
+=== Guidelines on content focus
+
+We want to help users to create Clean Code. Rule content should focus on the manner in which the code is not clean, why this is an issue, and how to remedy this. 
+Rules should talk about the potential impact on software quality in the 'What is the potential impact?' sub-section.
```
[the remark this answers is not in the document]
Comment (John-Clifton-SonarSource) [reply docs/description.adoc:25]: That's a good point. I have tried to add an example to clarify things a bit.

File: docs/description.adoc [hunk 2]
```diff
@@ -31,7 +49,7 @@ The possible sections for this format are the following:
 
 Third-level and fourth-level titles (`+=== Title t3+` and `+==== Title t4+`) are not checked for this type of rule.
 
-=== 2. Education Format
+=== 2. Educational Rule Format
```
[the remark this answers is not in the document]
Comment (John-Clifton-SonarSource) [reply docs/description.adoc]: Yes, updated.

File: docs/description.adoc [hunk 3]
```diff
@@ -6,6 +6,24 @@ This document describes how `+rule.adoc+` and its dependencies should be structu
...
+The goal is to provide the right level of guidance so that users get the right information from the rules at the right time.
+Do not feel obliged to use every section or sub-section if omitting them would lead to a better user experience.
+
+=== Guidelines if you aren’t sure where something belongs in a rule
+* If you need the information to fix the issue, but only the first time, it probably belongs in ‘Why is this an issue?’
+* If you need the information to fix the issue every time, it probably belongs in ‘How to fix it?’
+* If you don’t need the information to fix the issue, but it will help users grow their knowledge, it probably belongs in ‘Resources’ (this maps to the 'More Info' tab in the products)
+
+By being careful about what goes where, we help to ensure that users get exactly what they need, when they need it.
+
```
[the remark this answers is not in the document]
Comment (John-Clifton-SonarSource) [reply docs/description.adoc]: Agreed, I have moved it as suggested.

File: docs/description.adoc [hunk 4]
```diff
@@ -79,14 +79,14 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
 what is going on. This can be in the form of noncompliant and compliant code in a single code box (noncompliant lines should always be highlighted with the corresponding comment
-“// Noncompliant” optionally followed by some explanation) if that is clearer. This first tab will be like a blog post style with a free format explaining what the rule is
+“// Noncompliant” optionally followed by some explanation) if that is clearer. This first tab could use a blog post style with a free format explaining what the rule is
 detecting and why. Feel free to use “What is the potential impact?” title if it makes sense, or any other titles you find useful. +
-Goal: Understand the concepts behind an issue.
+Goal: Understand the concepts behind an issue and why it matters.
 +
 ** *What is the potential impact?* (level 3 title) [Optional]
 +
 What is the risk for me? What can go wrong?
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:88]:
```suggestion
This subsection aims to address questions such as the following:
* What is the risk for me?
* What is the risk for my organization?
* What is the risk for the users?
* What can go wrong?
```

File: docs/description.adoc [hunk 5]
```diff
@@ -79,14 +79,14 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
-This section can include multiple concrete threats as well.
+This section can include multiple concrete threats as well. This is the place to talk about the impacts to the software(e.g. security, maintainability, etc.)
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:89]:
```suggestion
This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see [docs/metadata#impacts](https://github.com/SonarSource/rspec/blob/master/docs/metadata.adoc#code-field).)*

Consequently, this section can also talk about the impacts on the ecosystem of this software. For example, impacts on the organization, on the users, and impact in terms of regulations.
```

File: docs/description.adoc [hunk 6]
```diff
@@ -217,6 +214,27 @@ This is important, as this format will be expected by the analyzers when loading
 Furthermore, the display name of the framework has to match an allowed framework
 display name, as defined in <<header_names/allowed_framework_names.adoc#,this allowed framework names file>>.
 
+==== General guidance
+
+Most sections and subsections of the Learn as You Code rule format are optional, only the `Why is this an issue?` main section is mandatory. 
+The goal is to provide the right level of guidance so that users get the right information from the rules at the right time.
+Do not feel obliged to use every section or sub-section if omitting them would lead to a better user experience.
+
+===== Guidelines if you aren’t sure where something belongs in a rule
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:223]:
```suggestion
==== Guidelines if you aren’t sure where something belongs in a rule
```

File: docs/description.adoc [hunk 7]
```diff
@@ -217,6 +214,27 @@ This is important, as this format will be expected by the analyzers when loading
...
+* If you need the information to fix the issue, but only the first time, it probably belongs in ‘Why is this an issue?’
+* If you need the information to fix the issue every time, it probably belongs in ‘How to fix it?’
+* If you don’t need the information to fix the issue, but it will help users grow their knowledge, it probably belongs in ‘Resources’ (this maps to the 'More Info' tab in the products)
+
+By being careful about what goes where, we help to ensure that users get exactly what they need, when they need it.
+
+===== Guidelines on content focus
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:230]:
```suggestion
==== Guidelines on content focus
```

**Review by loris-s-sonarsource — COMMENTED** [review]
I added two kinds of suggestion:
- clarity about the impact subsection, and going a little more into taxonomy details
- file headings, to avoid too much subsections

File: docs/description.adoc [hunk 8]
```diff
@@ -79,14 +79,17 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
 detecting and why. Feel free to use “What is the potential impact?” title if it makes sense, or any other titles you find useful. +
-Goal: Understand the concepts behind an issue.
+Goal: Understand the concepts behind an issue and why it matters.
 +
 ** *What is the potential impact?* (level 3 title) [Optional]
 +
-What is the risk for me? What can go wrong?
-This section can include multiple concrete threats as well.
+This subsection aims to address questions such as the following: +
+What is the risk for me?  What is the risk for my organization? What is the risk for the users? What can go wrong?
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:89]:
```suggestion
What is the risk for me?  What is the risk for my organization? What is the risk for the users? What can go wrong? +
```

File: docs/description.adoc [hunk 9]
```diff
@@ -79,14 +79,17 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
-Goal: Understand the concepts behind an issue.
+Goal: Understand the concepts behind an issue and why it matters.
 +
 ** *What is the potential impact?* (level 3 title) [Optional]
 +
-What is the risk for me? What can go wrong?
-This section can include multiple concrete threats as well.
+This subsection aims to address questions such as the following: +
+What is the risk for me?  What is the risk for my organization? What is the risk for the users? What can go wrong? +
+This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see [docs/metadata#impacts](https://github.com/SonarSource/rspec/blob/master/docs/metadata.adoc#code-field).)*
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:90]:
```suggestion
This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see <<metadata.adoc#,impacts>>.)*
```

File: docs/description.adoc [hunk 10]
```diff
@@ -79,14 +79,17 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
-Goal: Understand the concepts behind an issue.
+Goal: Understand the concepts behind an issue and why it matters.
 +
 ** *What is the potential impact?* (level 3 title) [Optional]
 +
-What is the risk for me? What can go wrong?
-This section can include multiple concrete threats as well.
+This subsection aims to address questions such as the following: +
+What is the risk for me?  What is the risk for my organization? What is the risk for the users? What can go wrong? +
+This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see <<metadata.adoc#,impacts>>.)*
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:90]:
```suggestion
This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see <<metadata.adoc#,code-field>>.)*
```

File: docs/description.adoc [hunk 11]
```diff
@@ -79,14 +79,17 @@ Goal: The title (or message) of a rule might not always be clear due to its shor
...
-Goal: Understand the concepts behind an issue.
+Goal: Understand the concepts behind an issue and why it matters.
 +
 ** *What is the potential impact?* (level 3 title) [Optional]
 +
-What is the risk for me? What can go wrong?
-This section can include multiple concrete threats as well.
+This subsection aims to address questions such as the following: +
+What is the risk for me?  What is the risk for my organization? What is the risk for the users? What can go wrong? +
+This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. *(see <<metadata.adoc#,code-field>>.)*
```
Comment (loris-s-sonarsource) [inline docs/description.adoc:90]:
```suggestion
This section aims to talk about the impacts on the software, depending on the code qualities this issue is linked to: security, reliability, or maintainability. **(see <<metadata.adoc#code-field,Impacts>>.)**
```

**Review by loris-s-sonarsource — APPROVED** [review]
LGTM ! you can merge it

**Pull request merged**